### DISCUSSION TITLE
You can refer to this framework.

### DISCUSSION THREAD

**Discussion by Allin2000** [nick-roberson/fastapi-gen#10] [General]
Loco is a Rails inspired web framework for Rust 。https://github.com/loco-rs/loco。

**Comment by Allin2000** [comment]
Axum has 16,000 stars. Fastapi has 70,000 stars.This means that if there is a Loco written in Python, it will have 15,000 stars. Ruby on Rails has 50000 stars.

**Reply by nick-roberson** [reply]
I dont understand what you are trying to tell me here....

**Reply by Allin2000** [reply]
Your framework must integrate the advantages of well-known frameworks to be popular

**Comment by Allin2000** [comment]
Loco follows Rails. There, I said it. Rails concepts are carefully adapted to modern Rust development.  （Ruby on Rails）  1.You can choose from different templates.   （advantage）2.You can perform database migration through the command line.  that's the Loco CLI. Generate code, operate your project.（django CLI？）  3.You have a variant of the MVC model, which removes the paradox of option.              What I mean is that you should look into this framework.What if Rails was Built on Python? Fastapi need a tools like loco.It's almost a rewrite of Hyperstackjs.io, and Hyperstack is based on an internal Rails-like framework which is production ready.

**Reply by nick-roberson** [reply]
Ah I see what you are saying here, I have some thoughts as a result. Basically enabling far more fine grained control of the downstream application from the FastAPI Gen CLI tool. 

I think that if I want to go down this path I really want to remove the compatibility with Mongo for the time being just to make development more singularly focused on support for Alembic.

**Reply by Allin2000** [reply]
When I searched for fastapi on github, there were a lot of template builds.If it's just a template, it's hard to succeed.I think your template needs to refer to some well-known frameworks. If your template references Django, FastAPI, Ruby on Rails. It must be very easy to become famous.Of course, it's very difficult. But you can let him have the advantages of loco first.

**Reply by Allin2000** [reply]
1.Users can choose what to build
 lightweight-service (minimal, only controllers and views)
  Rest API (with DB and user auth)
SaaS app (with DB and user auth)
2.Models are automatically generated through CLI tools
3.You have a variant of the MVC model, which removes the paradox of option. You deal with building your app, not making academic decisions for what abstractions to use.

...  Start simple.     If it has these advantages. It's bound to go beyond Django

**Reply by Allin2000** [reply]
hyperstackjs  Node.js
loco.rs  rust
These 2 are the same web framework. But it's written in a different programming language.

**Comment by nick-roberson** [comment]
All great ideas, will take into account, could you suggest some examples of additional features or whatnot other than DB management that I could add?

**Reply by Allin2000** [reply]
fastapi-gen:Inspired by ruby on rails ， based on fastapi, Better than Django。
Your template isn't currently inspired by Ruby on Rails.It should inlcudes almost every Rails feature with best-effort python ergonomics.

first: pip install fastapi-gen  or poetry install fastapi-gen?
You can select different templates from the command line.
Automatically generate models, CRUD, and perform database migrations from the command line.
Different deployment templates can be selected from the command line.
The configuration should be modifiable. For example, database settings.

**Comment by Allin2000** [comment]
I watched your video. Your template is fully functional. What may be lacking is a combinatorial design.What is the best architecture for FastAPI?fastapi is too concise.Your template should be designed to be the best FastAPI architecture？

**Comment by Allin2000** [comment]
There is a great advantage to having the option of choosing different templates. Everyone needs to choose a different template when building different projects.Provide 3-5 best fastapi templates for different situations.
